### PR TITLE
ICU-22350 Set minimal workflow token permissions

### PR DESCRIPTION
**Pull request by pnacht**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22350
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

As described in the issue, this PR sets read-only tokens for all workflows. Only the deployment job in `jekyll-gh-pages.yml` workflow receives a few additional permissions it needs.

This protects the project from supply-chain attacks.

### CODE REVIEW

**Review by echeran — APPROVED** [review]
LGTM, thanks for doing this!

**Review by markusicu — APPROVED** [review]
lgtm2 tnx!

I will propose to the team that we move the ticket to "accepted" status :-)

**Review by mihnita — APPROVED** [review]

**Review by mihnita — APPROVED** [review]
Thank you!
M.

**Review by FrankYFTang — APPROVED** [review]
LGTM

**Comment by markusicu** [comment]
> It looks as if this change broke all performance tests on the CI. Is that so?

Possible. We didn't look at those last week. They do write to a repo.

**Comment by pnacht** [comment]
Woah, sorry about that!

I'm taking a look at the code, is the issue the [icu_merge_ci.yml](https://github.com/unicode-org/icu/blob/2e3e49fed507108933c8b8ec07be2116b0722ecb/.github/workflows/icu_merge_ci.yml)? Specifically here (and similar):

https://github.com/unicode-org/icu/blob/2e3e49fed507108933c8b8ec07be2116b0722ecb/.github/workflows/icu_merge_ci.yml#L102-L115

I'm looking at that Action's documentation and I'm having a bit of trouble understanding which permissions it requires... it seems to save the results to `icu4c/source/test/perf/results/${{ matrix.perf }}/output.txt`, but the repo doesn't have a `.../results` folder...

Let me know and I'll send a PR to fix this ASAP. (Can I re-use the ICU-22350 bug ID?)

File: .github/workflows/maven.yaml [hunk 1]
```diff
@@ -2,6 +2,10 @@ name: Publish icu4j.jar/utilities.jar to GH Maven
 on:
   release:
     types: [created]
+
+permissions:
+  contents: read
```
Comment (srl295) [inline .github/workflows/maven.yaml:7]: this workflow needs to `mvn deploy` to github packages

Comment (srl295) [reply .github/workflows/maven.yaml:7]: in fact that's the only purpose of this workflow

Comment (srl295) [reply .github/workflows/maven.yaml:7]: so i'd just revert these two lines.

Comment (pnacht) [reply .github/workflows/maven.yaml:7]: Sorry! Fixed in https://github.com/unicode-org/icu/pull/2529.

**Pull request merged**